applications/GeoMechanicsApplication/python_scripts/geomechanics_T_solver.py

```diff
@@ -0,0 +1,209 @@
+            "problem_domain_sub_model_part_list": [""],
+            "processes_sub_model_part_list": [""],
+            "body_domain_sub_model_part_list": [""],
+            "loads_sub_model_part_list": [],
+            "loads_variable_list": []
+        }""")
+
+        this_defaults.AddMissingParameters(super().GetDefaultParameters())
+        return this_defaults
+
+    # =============================================================================================
+    # =============================================================================================
+    def PrepareModelPart(self):
+        super().PrepareModelPart()
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "Model reading finished.")
+
+    # =============================================================================================
+    # =============================================================================================
+    def AddDofs(self):
+        ## Fluid dofs
+        KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.TEMPERATURE, self.main_model_part)
+
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "DOFs added correctly.")
+
+    # =============================================================================================
+    # =============================================================================================
+    def Initialize(self):
+        KratosMultiphysics.Logger.PrintInfo("::[GeoMechanics_T_Solver]:: ", "Initialisation ...")
+        
+        super().Initialize()
+
+        self.find_neighbour_elements_of_conditions_process = KratosGeo.FindNeighbourElementsOfConditionsProcess(self.computing_model_part)
+        self.find_neighbour_elements_of_conditions_process.Execute()
+
+        self.deactivate_conditions_on_inactive_elements_process = KratosGeo.DeactivateConditionsOnInactiveElements(self.computing_model_part)
+        self.deactivate_conditions_on_inactive_elements_process.Execute()
+
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "solver.Initialize is set successfully")
+
+        # Check if everything is assigned correctly
+        self.Check()
+
+        KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver", "Solver initialization finished.")
+
+
+    #### Specific internal functions ####
+
+    # =============================================================================================
+    # =============================================================================================
+    def _ConstructScheme(self, scheme_type, solution_type):
+
+        self.main_model_part.ProcessInfo.SetValue(KratosGeo.DT_TEMPERATURE_COEFFICIENT, 1.0)
+
+        if (scheme_type.lower() == "newmark" or scheme_type.lower() == "newmark_flow"):
+            theta = self.settings["newmark_theta"].GetDouble()
+            rayleigh_m = self.settings["rayleigh_m"].GetDouble()
+            rayleigh_k = self.settings["rayleigh_k"].GetDouble()
```

Comment:
Not relevant, there is no damping matrix constructed from mass and stiffness matrices in a T solver

Reply:
Removed